### PR TITLE
Adding Basic Headings PR NUMBER TWO

### PR DESCRIPTION
**Pull request by hrmeiners**
NOTE: this code has not been tested, as I am still working on getting a Windows virtual machine or parallels in place.

Fixes issue https://github.com/oss-slu/PilotDataSynchronization/issues/15
What was changed?
--- Copied from the 'get_altitudes.cpp' file that Ali Elnour originally wrote, this is a basic plugin initialization that deals with the Heading DataRefs for the Pilot and CoPilot's instruments.
--- Nothing in his original code has been changed except for the DataRef names and a function name (that of drawHelloWorld).

Why was it changed?
--- The code did not exist beforehand, so this is the first step in extracting the data.

How was it changed?
--- The file was created
--- The XPluginStart() function was changed to include the new data refs and variable names for the Headings
--- DrawHelloWorld() function was changed to draw_heading_window(), had the output strings changed to fit the new file, and the data it prints was changed to be the Heading DataRefs.

This is a new pull request since for some reason, my changes to the old pull request and my branch were not committing or pushing. The old PR can be seen here: #18 

This PR has a new commit that is focused on fixing @harshitha-thota 's bug she encountered in Line 29, where I attempted to comment out some information and accidentally made it an "ifndef ____" statement. There are also some minor variable changes included in this PR

### CODE REVIEW

**Comment by harshitha-thota** [comment]
I got below errors:

src/get_heading.cpp(66): error C2065: 'draw_hello_world': undeclared identifier
src/get_heading.cpp(92): error C2065: 'g_window': undeclared identifier
src/get_heading.cpp(94): error C2065: 'g_window': undeclared identifier
src/get_heading.cpp(95): error C2065: 'g_window': undeclared identifier
src/get_heading.cpp(101): error C2065: 'g_window': undeclared identifier
src/get_heading.cpp(123): error C2665: 'std::to_string': no overloaded function could convert all the argument types
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(506): note: could be 'std::string std::to_string(long double)'
src/get_heading.cpp(123): note: 'std::string std::to_string(long double)': cannot convert argument 1 from 'XPLMDataRef' to 'long double'
src/get_heading.cpp(123): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(502): note: or       'std::string std::to_string(float)'
src/get_heading.cpp(123): note: 'std::string std::to_string(float)': cannot convert argument 1 from 'XPLMDataRef' to 'float'
src/get_heading.cpp(123): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(495): note: or       'std::string std::to_string(double)'
src/get_heading.cpp(123): note: 'std::string std::to_string(double)': cannot convert argument 1 from 'XPLMDataRef' to 'double'
src/get_heading.cpp(123): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(491): note: or       'std::string std::to_string(unsigned __int64)'
src/get_heading.cpp(123): note: 'std::string std::to_string(unsigned __int64)': cannot convert argument 1 from 'XPLMDataRef' to 'unsigned __int64'
src/get_heading.cpp(123): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(487): note: or       'std::string std::to_string(__int64)'
src/get_heading.cpp(123): note: 'std::string std::to_string(__int64)': cannot convert argument 1 from 'XPLMDataRef' to '__int64'
src/get_heading.cpp(123): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(483): note: or       'std::string std::to_string(unsigned long)'
src/get_heading.cpp(123): note: 'std::string std::to_string(unsigned long)': cannot convert argument 1 from 'XPLMDataRef' to 'unsigned long'
src/get_heading.cpp(123): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(479): note: or       'std::string std::to_string(long)'
src/get_heading.cpp(123): note: 'std::string std::to_string(long)': cannot convert argument 1 from 'XPLMDataRef' to 'long'
src/get_heading.cpp(123): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(475): note: or       'std::string std::to_string(unsigned int)'
src/get_heading.cpp(123): note: 'std::string std::to_string(unsigned int)': cannot convert argument 1 from 'XPLMDataRef' to 'unsigned int'
src/get_heading.cpp(123): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(471): note: or       'std::string std::to_string(int)'
src/get_heading.cpp(123): note: 'std::string std::to_string(int)': cannot convert argument 1 from 'XPLMDataRef' to 'int'
src/get_heading.cpp(123): note: There is no context in which this conversion is possible
src/get_heading.cpp(123): note: while trying to match the argument list '(XPLMDataRef)'
src/get_heading.cpp(123): error C2088: built-in operator '+' cannot be applied to an operand of type 'std::string'
src/get_heading.cpp(125): error C2665: 'std::to_string': no overloaded function could convert all the argument types
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(506): note: could be 'std::string std::to_string(long double)'
src/get_heading.cpp(125): note: 'std::string std::to_string(long double)': cannot convert argument 1 from 'XPLMDataRef' to 'long double'
src/get_heading.cpp(125): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(502): note: or       'std::string std::to_string(float)'
src/get_heading.cpp(125): note: 'std::string std::to_string(float)': cannot convert argument 1 from 'XPLMDataRef' to 'float'
src/get_heading.cpp(125): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(495): note: or       'std::string std::to_string(double)'
src/get_heading.cpp(125): note: 'std::string std::to_string(double)': cannot convert argument 1 from 'XPLMDataRef' to 'double'
src/get_heading.cpp(125): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(491): note: or       'std::string std::to_string(unsigned __int64)'
src/get_heading.cpp(125): note: 'std::string std::to_string(unsigned __int64)': cannot convert argument 1 from 'XPLMDataRef' to 'unsigned __int64'
src/get_heading.cpp(125): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(487): note: or       'std::string std::to_string(__int64)'
src/get_heading.cpp(125): note: 'std::string std::to_string(__int64)': cannot convert argument 1 from 'XPLMDataRef' to '__int64'
src/get_heading.cpp(125): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(483): note: or       'std::string std::to_string(unsigned long)'
src/get_heading.cpp(125): note: 'std::string std::to_string(unsigned long)': cannot convert argument 1 from 'XPLMDataRef' to 'unsigned long'
src/get_heading.cpp(125): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(479): note: or       'std::string std::to_string(long)'
src/get_heading.cpp(125): note: 'std::string std::to_string(long)': cannot convert argument 1 from 'XPLMDataRef' to 'long'
src/get_heading.cpp(125): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(475): note: or       'std::string std::to_string(unsigned int)'
src/get_heading.cpp(125): note: 'std::string std::to_string(unsigned int)': cannot convert argument 1 from 'XPLMDataRef' to 'unsigned int'
src/get_heading.cpp(125): note: There is no context in which this conversion is possible
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\string(471): note: or       'std::string std::to_string(int)'
src/get_heading.cpp(125): note: 'std::string std::to_string(int)': cannot convert argument 1 from 'XPLMDataRef' to 'int'
src/get_heading.cpp(125): note: There is no context in which this conversion is possible
src/get_heading.cpp(125): note: while trying to match the argument list '(XPLMDataRef)'
src/get_heading.cpp(125): error C2088: built-in operator '+' cannot be applied to an operand of type 'std::string'

File: src/pilotdatasync-xp11.cpp [hunk 1]
```diff
@@ -31,13 +31,16 @@ DllMain(HANDLE hModule, DWORD ul_reason_for_call, LPVOID lpReserved) {
 // An opaque handle to the window we will create
 static XPLMWindowID g_window;
```
Comment (carlyrhoover) [inline src/pilotdatasync-xp11.cpp:32]: Changes in function names. Should be ok, but could raise some errors

**Review by carlyrhoover — COMMENTED** [review]
Remove binary file

**Comment by harshitha-thota** [comment]
Suggested Improvements:

1. I see that the plugin extracts both the pilot and co-pilot heading values, but we need to focus only on the pilot headings value.

2. Add error handling for scenarios where the heading data references might not retrieve valid data. This will improve the robustness of the plugin.
 
3. Remove the .DS_Store file.

Next Steps:
Please address these suggestions, and once you’ve made the improvements, I will review the updated code. After completing my review and testing, I look forward to merging the PR if everything works as expected.

Keep up the great work!

File: utils/init.ps1
Comment (harshitha-thota) [inline utils/init.ps1:1]: New-Item ../lib/SDK -ItemType Directory -Force
Expand-Archive $download_loc -DestinationPath "../lib/" -Force

In the above 2 lines you mentioned (..) two dots. Revert back these changes so, that the lib folder will be downloaded in the correct location.

**Review by harshitha-thota — APPROVED** [review]

**Pull request merged**